### PR TITLE
Distinct events

### PR DESCRIPTION
**Pull request by maxaalexeeva**
A first stab at separating Assignment events into more fine-grained categories:
- rules split and some new ones added (including a few for finding crops and fertilizers that have not been found with an ner)
- some changes to tests: passing labels as an arg to tests + a test for empty sequences
- updating methods to not assume the Assignment label

### CODE REVIEW

File: src/main/scala/org/clulab/habitus/actions/HabitusActions.scala [hunk 1]
```diff
@@ -152,4 +163,29 @@ class HabitusActions extends Actions {
     }
     toReturn
   }
+
+  def FertilizerEventToRelation(mentions: Seq[Mention]): Seq[Mention] = {
+    // creates a binary fertilizer event from a rule where the trigger was a token that should also serve as a variable in an assignment event
+    val toReturn = new ArrayBuffer[Mention]()
+    for (m <- mentions) {
```
Comment (kwalcock) [inline src/main/scala/org/clulab/habitus/actions/HabitusActions.scala:170]: This looks like a straight 1-to-1 conversion, so  a map:
```scala
    mentions.map { m =>
      ....
      new RelationMention(
        ...
      )
    }
```

File: src/test/scala/org/clulab/habitus/variables/TestVariableReaderLabelBased.scala [hunk 2]
```diff
@@ -16,34 +16,44 @@ class TestVariableReaderLabelBased extends Test {
...
           println("Put a breakpoint here to observe a particular test.")
 
-        val mentions = getMentions(text).filter(_.label matches "Assignment")
+        val allMentions = getMentions(text)
+        val mentions = allMentions.filter(_.label matches label).sortBy(_.tokenInterval)
 
-        mentions should have size variables.length
+        if (variables.isEmpty) {
+          // get only relations and events
+          val nonTextBoundMentions = mentions.filter(m => !m.isInstanceOf[TextBoundMention])
```
Comment (kwalcock) [inline src/test/scala/org/clulab/habitus/variables/TestVariableReaderLabelBased.scala:38]: Some places have a rule that this has to be .filterNot.  (It's on a best practices list somewhere.)

**Review by kwalcock — APPROVED** [review]
I'm not able to review the rules.  None of the other comments should result in changed functionality.  It's more that whenever the code is touched, it should get a little bit nicer when possible.

File: src/main/scala/org/clulab/habitus/actions/HabitusActions.scala [hunk 3]
```diff
@@ -109,15 +111,24 @@ class HabitusActions extends Actions {
...
-    val (targetMentions, other) = mentions.partition(_.label == "Assignment")
+//    val (targetMentions, other) = mentions.partition(_.label == "Assignment")
+    // the action applies to mentions extracted with var reader relation/event rules, so we exclude tbms and include a value argument
+    val (targetMentions, other) = mentions.partition(m => !m.isInstanceOf[TextBoundMention] & m.arguments.keys.toList.contains("value"))
+
+
     val groupedBySent = targetMentions.groupBy(_.sentence)
     for (sentGroup <- groupedBySent) {
-        val groupedByValue = sentGroup._2.groupBy(_.arguments("value").head.text)
+      // only pick out of those with the same label
```
[the remark this answers is not in the document]
Comment (maxaalexeeva) [reply src/main/scala/org/clulab/habitus/actions/HabitusActions.scala]: @kwalcock thanks! Is this an efficiency issue, readability issue, something else?

File: src/main/scala/org/clulab/habitus/actions/HabitusActions.scala (hunk 1 above)
Comment (maxaalexeeva) [reply src/main/scala/org/clulab/habitus/actions/HabitusActions.scala]: that does look much better!

**Comment by maxaalexeeva** [comment]
@kwalcock @MihaiSurdeanu updated based on your feedback

File: src/main/scala/org/clulab/habitus/actions/HabitusActions.scala [hunk 4]
```diff
@@ -152,4 +163,29 @@ class HabitusActions extends Actions {
     }
     toReturn
   }
+
+  def FertilizerEventToRelation(mentions: Seq[Mention]): Seq[Mention] = {
```
[the remark this answers is not in the document]
Comment (kwalcock) [reply src/main/scala/org/clulab/habitus/actions/HabitusActions.scala]: Oh, now I see why the uppercase.  The Scala conventions are a little at odds with the grammar ones.  I learned something!

File: src/main/scala/org/clulab/habitus/actions/HabitusActions.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (maxaalexeeva) [reply src/main/scala/org/clulab/habitus/actions/HabitusActions.scala]: I am learning! That's good! Thanks!

[the remark this answers is not in the document]
Comment (maxaalexeeva) [reply src/main/scala/org/clulab/habitus/actions/HabitusActions.scala]: @IkeKobby when you have a minute, have a look at how @kwalcock simplified by array buffer code here. Related to what we discussed about array buffers and maps the other day.

File: src/main/scala/org/clulab/habitus/actions/HabitusActions.scala (hunk 4 above)
[the remark this answers is not in the document]
Comment (kwalcock) [reply src/main/scala/org/clulab/habitus/actions/HabitusActions.scala]: I checked eidos and reach and so far have only found lowercase actions, so I'd stick with that if there aren't counterexamples.  Some eventual maintainer will be thankful for one less curve ball.  I had just forgotten the need for coordination between the files.

**Comment by maxaalexeeva** [comment]
@kwalcock @MihaiSurdeanu safe to merge?

**Comment by kwalcock** [comment]
LGTM

**Comment by MihaiSurdeanu** [comment]
LGTM too!

**Comment by maxaalexeeva** [comment]
@IkeKobby fyi

**Comment by IkeKobby** [comment]
@maxaalexeeva  okay.

**Pull request merged**